ksqldb-engine/src/main/java/io/confluent/ksql/internal/ScalablePushQueryExecutorMetrics.java

```diff
@@ -0,0 +1,728 @@
+      rowsProcessedSensorMap.get(key).record(0);
+    } else {
+      throw new IllegalStateException("Metrics not configured correctly, missing " + key);
+    }
+  }
+
+  public List<Sensor> getSensors() {
+    return Collections.unmodifiableList(sensors);
+  }
+
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "should be mutable")
+  public Metrics getMetrics() {
+    return metrics;
+  }
+
+  private Sensor configureLocalRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        SCALABLE_PUSH_QUERY_METRIC_GROUP + "-" + SCALABLE_PUSH_REQUESTS + "-local");
+
+    // legacy
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-local-count",
+        ksqlServiceIdLegacyPrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Count of local scalable push query requests",
+        legacyCustomMetricsTags,
+        new CumulativeCount()
+    );
+
+    // new metrics with ksql service id in tags
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-local-count",
+        ksqlServicePrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Count of local scalable push query requests",
+        customMetricsTags,
+        new CumulativeCount()
+    );
+
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureRemoteRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        SCALABLE_PUSH_QUERY_METRIC_GROUP + "-" + SCALABLE_PUSH_REQUESTS + "-remote");
+
+    // legacy
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-remote-count",
+        ksqlServiceIdLegacyPrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Count of remote scalable push query requests",
+        legacyCustomMetricsTags,
+        new CumulativeCount()
+    );
+
+    // new metrics with ksql service in tags
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-remote-count",
+        ksqlServicePrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Count of remote scalable push query requests",
+        customMetricsTags,
+        new CumulativeCount()
+    );
+
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureErrorRateSensor() {
+    final Sensor sensor = metrics.sensor(
+        SCALABLE_PUSH_QUERY_METRIC_GROUP + "-" + SCALABLE_PUSH_REQUESTS + "-error-rate");
+    // legacy
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-error-rate",
+        ksqlServiceIdLegacyPrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Rate of erroneous scalable push query requests",
+        legacyCustomMetricsTags,
+        new Rate()
+    );
+    addSensor(
+        sensor,
+        SCALABLE_PUSH_REQUESTS + "-error-total",
+        ksqlServiceIdLegacyPrefix + SCALABLE_PUSH_QUERY_METRIC_GROUP,
+        "Total number of erroneous scalable push query requests",
+        legacyCustomMetricsTags,
+        new CumulativeCount()
+    );
+
+    // new metrics with ksql service id in tags
+    addSensor(
```

Comment:
If we're going to have a rate sensor for errors, we should also have it for requests as well.  I know I was talking about the use of these vs using totals and their timestamps and favoring the latter.  It's probably best to emit both rate and total metrics and we can leave it to the consumer to decide which is preferable to use.